src/Select/Select.tsx

```diff
@@ -57,11 +57,12 @@ function Select<T extends string | number | readonly string[] | undefined>({
         {...(value && { value })}
         sx={sx}
       >
-        {values.map(({ value, text }) => (
+        {values.map(({ value, text, disabled }) => (
           <MenuItem
             key={buildOptionId?.(value)}
             id={buildOptionId?.(value)}
             value={value}
+            disabled={Boolean(disabled)}
```

Comment:
Actually I don't think you need to Cast it to a Boolean as the `boolean | undefined` type of the argument already matches the behaviour of the `disabled` prop. Feel free to keep it however, if you think it makes it more clear.